### ISSUE TITLE
Unused eslint-disable comments should warn when it does not apply to the corresponding code

### ISSUE THREAD

**Issue by growlsworth** [eslint/eslint#4971]
This is kinda a meta enhancement. It would be a nice violation for any `eslint-disable` rules to actually match a violation that exists in the corresponding code. That way there are no vestigial `eslint-disable` comments. `eslint-disable` is usually added when some offending code "cannot" be changed at the moment, and then eventually the offending code is removed, and the `eslint-disable` remains. I know it's important to minimize the usage of `eslint-disable` in the first place, and I think this will help ensure that the uses are still valid.

Thanks!

**Comment by eslintbot** [comment]
@growlsworth Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by growlsworth** [comment]
Thanks @eslintbot . 

1. I'm using `v1.10.3`
2. config:
```
{
  "extends": ["airbnb"], // "eslint-config-airbnb": "3.1.0", "eslint-plugin-react": "3.15.0",
  "env": {
    "es6": true
  }
}
```

**Comment by growlsworth** [comment]
1. Sample code:
```js
/* eslint-disable no-console  */
```
Sample output:
```sh
$ eslint sample.js
> 1:1  warning  Unnecessary eslint-disable statement no-console
```
2. It does not prevent a syntax nor programming error.
3. I think it should be in core because it could apply to all rules.
4. I haven't created a rule before so I wouldn't be especially quick at it. I wouldn't start on it unless the core community thought it was a good idea.

**Comment by ilyavolodin** [comment]
This would be extremely hard to do, since currently disabled rule is not even running between comments.

**Comment by nzakas** [comment]
Agreed. It's an interesting idea, but I think the additional complexity isn't worth the incremental value at this point.

**Comment by growlsworth** [comment]
Thanks for considering it.

**Comment by alecxe** [comment]
Cross-referencing SO thread: [Rule is disabled but not violated](http://stackoverflow.com/questions/36895440/rule-is-disabled-but-not-violated). Would appreciate any ideas or workarounds. Thanks!
